internal/models/interfaceconfig.go

```diff
@@ -0,0 +1,66 @@
+package models
+
+import (
+	"bytes"
+	"text/template"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+var (
+	tunnelTemplate = `[Interface]
+PrivateKey = {{.privateKey}}
+Address = {{.address}}
+
+{{range .interfacePeers}}
+[Peer]
+PublicKey = {{.PublicKey}}
+AllowedIPs = {{.AllowedIPs}}
+Endpoint = {{.Endpoint}}
+{{end}}`
+)
+
+// InterfaceConfig type represents a WireGuard tunnel configuration.
+type InterfaceConfig struct {
+	privateKey     string
+	address        string
+	interfacePeers []wgtypes.Peer
+	tmpl 		Template
+}
+
+// NewInterfaceConfig returns a new interfaceConfig.
+func NewInterfaceConfig() *InterfaceConfig {
+	newIconfig := InterfaceConfig{}
+	return &newIconfig
+}
+
+// init initializes a new InterfaceConfig.
+func (i *InterfaceConfig) init() {
```

Comment:
This shouldn't be bound on the type.  Since the template is a constant across the entire package, there should be a package scoped variable that gets this value assigned to it in a package scoped init().